docker-compose-example.yml

```diff
@@ -26,7 +26,9 @@ services:
       - ./examples/prom-statsd-exporter/conf.yaml:/etc/statsd-exporter/conf.yaml
 
   ratelimit:
-    image: envoyproxy/ratelimit:master
+    build:
+      context: .
+      dockerfile: Dockerfile
```

Comment:
while it makes sense to have a test env that uses latest local code, I think we should make another docker-compose file for that. once we merged this PR in, the `master` image should pick up all shadowmade changes. 

kind of separate but I was thinking if we had a docker compose env that builds off the latest code, we could use it to do basic integration testing + performance testing in GH Actions via curl and timing the response + checking status codes.

Reply:
I totally agree, that's what we typically do at my place. I'll put together an integration test and push

Reply:
I added a self-contained docker-compose with some simple end-to-end-tests.

Make file target will trigger the tests and exit with a non-0 code if any fails